src/cart/swim/swim.c

```diff
@@ -870,9 +858,9 @@ swim_progress(struct swim_context *ctx, int64_t timeout)
 		D_GOTO(out_err, rc = 0); /* Ignore this update */
 
 	now = swim_now_ms();
-	if (timeout > 0)
-		end = now + timeout;
-	ctx->sc_next_event = now + swim_period_get() / 3;
+	if (timeout_us > 0)
+		end = now + timeout_us / 1000; /* timeout in us */
+	ctx->sc_next_event = now + swim_period_get();
```

Comment:
could you explain why it's period()/2 and now is just period()?

Reply:
It was overcheck to definitely don't miss event. I encountered this didn't helped and use now precise values instead of /2 or /3.